src/NextjsRevalidation.ts

```diff
@@ -74,16 +87,83 @@ export class NextjsRevalidation extends Construct {
 
   private createFunction(): LambdaFunction {
     const commonFnProps = getCommonFunctionProps(this);
-    const fn = new LambdaFunction(this, 'Fn', {
+    const fn = new LambdaFunction(this, 'QueueFn', {
       ...commonFnProps,
       // open-next revalidation-function
       // see: https://github.com/serverless-stack/open-next/blob/274d446ed7e940cfbe7ce05a21108f4c854ee37a/README.md?plain=1#L65
       code: Code.fromAsset(this.props.nextBuild.nextRevalidateFnDir),
       handler: 'index.handler',
-      description: 'Next.js revalidation function',
+      description: 'Next.js Queue Revalidation Function',
       timeout: Duration.seconds(30),
     });
     fn.addEventSource(new SqsEventSource(this.queue, { batchSize: 5 }));
     return fn;
   }
+
+  private createRevalidationTable() {
+    return new Table(this, 'Table', {
+      partitionKey: { name: 'tag', type: AttributeType.STRING },
+      sortKey: { name: 'path', type: AttributeType.STRING },
+      billing: Billing.onDemand(),
+      globalSecondaryIndexes: [
+        {
+          indexName: 'revalidate',
+          partitionKey: { name: 'path', type: AttributeType.STRING },
+          sortKey: { name: 'revalidatedAt', type: AttributeType.NUMBER },
+        },
+      ],
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+
+  /**
+   * This function will insert the initial batch of tag / path / revalidation data into the DynamoDB table during deployment.
+   * @see: {@link https://open-next.js.org/inner_workings/isr#tags}
+   *
+   * @param revalidationTable table to grant function access to
+   * @returns the revalidation insert provider function
+   */
+  private createRevalidationInsertFunction(revalidationTable: Table) {
+    const dynamodbProviderPath = this.props.nextBuild.nextRevalidateDynamoDBProviderFnDir;
+
+    // note the function may not exist - it only exists if there are cache tags values defined in Next.js build meta files to be inserted
+    // see: https://github.com/sst/open-next/blob/c2b05e3a5f82de40da1181e11c087265983c349d/packages/open-next/src/build.ts#L426-L458
+    if (fs.existsSync(dynamodbProviderPath)) {
+      const commonFnProps = getCommonFunctionProps(this);
+      const insertFn = new LambdaFunction(this, 'DynamoDBProviderFn', {
+        ...commonFnProps,
+        // open-next revalidation-function
+        // see: https://github.com/serverless-stack/open-next/blob/274d446ed7e940cfbe7ce05a21108f4c854ee37a/README.md?plain=1#L65
+        code: Code.fromAsset(this.props.nextBuild.nextRevalidateDynamoDBProviderFnDir),
+        handler: 'index.handler',
+        description: 'Next.js Revalidation DynamoDB Provider',
+        timeout: Duration.minutes(15),
+        initialPolicy: [
```

Comment:
Instead of specifying an `initialPolicy`, what do you think about adding:
```
revalidationTable.grantReadWrite(insertFn);
```
below? Keeps it simpler IMO.